runtime_scan/pkg/orchestrator/discovery/scope.go

```diff
@@ -65,3 +64,74 @@ func (sd *ScopeDiscoverer) Start(ctx context.Context) {
 		}
 	}()
 }
+
+func (sd *ScopeDiscoverer) DiscoverAndCreateAssets(ctx context.Context) error {
+	discoveryTime := time.Now()
+
+	assetTypes, err := sd.providerClient.DiscoverTargets(ctx)
```

Comment:
maybe better to do it paginated not to blowup the memory when there are many assets (separate task)?

Reply:
thats a great point, the AWS DiscoverAssets flow paginates the AWS call but it still results in all the models.AssetInfo's in memory. I should probably change that into a channel so the pagination is passed through.

Reply:
https://github.com/openclarity/vmclarity/issues/445